Sources/NIO/ChannelPipeline.swift

```diff
@@ -662,8 +662,12 @@ public final class ChannelPipeline: ChannelInvoker {
         return eventLoop.inEventLoop
     }
 
-    // Only executed from Channel
-    init (channel: Channel) {
+    /// Create `ChannelPipeline` for a given `Channel`. Although this method is public, it's very unlikely that you
+    /// will need this unless you're implementing and use child `Channel`s.
```

Comment:
@weissi can you make this more clear that this should only be used for custom `Channel` implementations.

Something like:

Create a `ChannelPipeline` for a `Channel`.  This method _should_ never be called by the end-user and only be used for custom `Channel` implementations.  Users should always use `channel.pipeline` to access the `ChannelPipeline` of a `Channel`.

This way it is more clear imho.

WDYT ?